### PR TITLE
Adds functionality for updating cache entries

### PR DESCRIPTION
**Pull request by boehmseb**
Resolves se-passau/VaRA#429

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ varats/data/cache_helper.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5279101&bid=17285546)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/cache_helper.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5279101&bid=17285546)

File: varats/data/cache_helper.py [hunk 1]
```diff
@@ -12,7 +12,10 @@
 from varats.settings import CFG
 from varats.data.report import BaseReport, MetaReport
 
+LOG = logging.getLogger(__name__)
+
 CACHE_REVISION_COL = 'cache_revision'
+CACHE_TIMESTAMP_COL = 'cache_timestamp'
 # TODO (se-passau/VaRA#518): rename to plot
```
Comment (vulder) [inline varats/data/cache_helper.py:19]:
```suggestion
```

File: varats/data/cache_helper.py [hunk 2]
```diff
@@ -80,65 +83,117 @@ def cache_dataframe(data_id: tp.Union[GraphCacheType], project_name: str,
...
+
 def build_cached_report_table(
         graph_cache_type: GraphCacheType, project_name: str,
         create_empty_df: tp.Callable[[], pd.DataFrame],
         create_df_from_report: tp.Callable[[tp.Any], pd.DataFrame],
-        create_report: tp.Callable[[Path], BaseReport],
-        report_files: tp.List[Path]) -> pd.DataFrame:
+        create_report: tp.Callable[[Path],
+                                   BaseReport], report_files: tp.List[Path],
+        failed_report_files: tp.List[Path]) -> pd.DataFrame:
```
Comment (vulder) [inline varats/data/cache_helper.py:114]: why do you need the failed report files?

Comment (boehmseb) [reply varats/data/cache_helper.py:114]: I use them to determine which entries I have to drop from the cache. When I think about it now, I could probably infer this from the `report_files` (just drop all entries that do not occur there).

File: varats/data/cache_helper.py (hunk 1 above)
Comment (boehmseb) [reply varats/data/cache_helper.py:19]: Seems like my comment to this suggestion somehow got lost.
Anyway, despite this issue being closed, it is not fixed in the code here yet. However, I will resolve this during my rework of the data layer and would leave this in until then.

**Review by vulder — APPROVED** [review]

**Pull request merged**